File: ccif/src/java/other/org/himalay/ccif/CancelReq.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:57 EST 2016")

public  class CancelReq extends   APDUBaseFactory.APDUBase { //Concrete type is CancelReq




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // IPPVslotId
    public short IPPVslotId ;
    // pinCodeLength
    public ByteArray pinCodeLength ;

    public CancelReq () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize IPPVslotId

        // Initialize pinCodeLength
        pinCodeLength= new ByteArray();
        pinCodeLength.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read IPPVslotId
        {
            IPPVslotId=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read pinCodeLength
        {
            retVal+=pinCodeLength.read(istream);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read IPPVslotId
        {
            IPPVslotId=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read pinCodeLength
        {
            retVal+=pinCodeLength.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }


        {   /** fix dependent sizes for pinCodeLength**/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write IPPVslotId
        ostream.writeByte(IPPVslotId);
        retVal +=1;
        // write pinCodeLength
        {
            retVal += pinCodeLength.write(ostream);
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("CancelReq\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write IPPVslotId
        dc.indent();
        dc.getPs().println("IPPVslotId="+IPPVslotId+"(0x"+ Integer.toHexString(IPPVslotId) + ")" );
        // write pinCodeLength
        dc.indent();
        dc.getPs().print("pinCodeLength: "+pinCodeLength.getSize()+"(0x"+Integer.toHexString(pinCodeLength.getSize())+")\n");
        this.pinCodeLength.dump(dc);
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for IPPVslotId
    //public short getIPPVslotId()
    //{
    //    return IPPVslotId ;
    //}


    // Setter for IPPVslotId
    //public void setIPPVslotId(short val)
    //{
    //    this.IPPVslotId= val;
    //}
    // Getter for pinCodeLength
    //public ByteArray getPinCodeLength()
    //{
    //    return pinCodeLength ;
    //}


    // Setter for pinCodeLength
    //public void setPinCodeLength(ByteArray val)
    //{
    //    this.pinCodeLength= val;
    //}


    public void setPinCodeLength(byte[] val)
    {
        this.pinCodeLength.setData(val);
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/CancelReq.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:14 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "CancelReq.h"

#ifdef org_himalay_ccif_CancelReq__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
CancelReq::CancelReq() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // IPPVslotId
    IPPVslotId= (ui8)0;
    // pinCodeLength
    org_himalay_msgs_runtime_ByteArray_NEW(pinCodeLength);
    pinCodeLength->setSizeType("FIRST_UI8");
}
// Destructor
CancelReq::~CancelReq() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // IPPVslotId

    // pinCodeLength
    org_himalay_msgs_runtime_ByteArray_DELETE( pinCodeLength);
}

int CancelReq::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // IPPVslotId
    {
        IPPVslotId=(istream.readUI8());
        retVal+=1;
    }
    // pinCodeLength
    {
        retVal+=pinCodeLength->read(istream);
    }
    return retVal;
}

int CancelReq::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read IPPVslotId
    {
        IPPVslotId=(istream.readUI8());
        retVal+=1;
    }
    // read pinCodeLength
    {
        retVal+=pinCodeLength->read(istream);
    }

    return retVal;
}

int CancelReq::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }


    {   /** fix dependent sizes for pinCodeLength**/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write IPPVslotId
    ostream.writeUI8(IPPVslotId);
    retVal +=1;
    // write pinCodeLength
    {
        retVal += pinCodeLength->write(ostream);
    }

    return retVal;
}

int CancelReq::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("CancelReq\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write IPPVslotId
    dc.indent();
    dc.getPs().print("IPPVslotId=");
    dc.getPs().println((long)IPPVslotId);
    // write pinCodeLength
    dc.indent();
    dc.getPs().print("pinCodeLength");
    pinCodeLength->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE CancelReq::getHeader()
//{
//return this->header;
//}

// Setter for header
//void CancelReq::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 CancelReq::getLength()
//{
//return this->length;
//}

// Setter for length
//void CancelReq::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for IPPVslotId
//ui8 CancelReq::getIPPVslotId()
//{
//return this->IPPVslotId;
//}

// Setter for IPPVslotId
//void CancelReq::setIPPVslotId(ui8 val)
//{
//this->IPPVslotId= val;
//}

// Getter for pinCodeLength
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE CancelReq::getPinCodeLength()
//{
//return this->pinCodeLength;
//}

// Setter for pinCodeLength
//void CancelReq::setPinCodeLength(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->pinCodeLength= val;
//}

int CancelReq::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void CancelReq::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE CancelReq::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_CancelReq__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::CancelReq* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::CancelReq* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code